FreeRTOS/Test/CMock/queue/tracing/queue_registry_utest.c

```diff
@@ -0,0 +1,398 @@
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+/*! @file queue_registry_utest.c */
+
+/* C runtime includes. */
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+
+#include "../queue_utest_common.h"
+
+/* Queue includes */
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "queue.h"
+
+/* ============================  GLOBAL VARIABLES =========================== */
+
+/**
+ * @brief Copy of QueueRegistryItem_t from queue.c to allow clearing items between test cases.
+ */
+typedef struct QUEUE_REGISTRY_ITEM
+{
+    const char * pcQueueName;
+    QueueHandle_t xHandle;
+} xQueueRegistryItem;
+typedef xQueueRegistryItem QueueRegistryItem_t;
+
+/* Access the xQueueRegistry to clear between test cases */
+extern PRIVILEGED_DATA QueueRegistryItem_t xQueueRegistry[ configQUEUE_REGISTRY_SIZE ];
+
+/* ==========================  CALLBACK FUNCTIONS =========================== */
+
+/* ==========================  Unity fixtures =========================== */
+
+void setUp( void )
+{
+    commonSetUp();
+    /* Clear the queue registry between test cases */
+    memset( &xQueueRegistry, 0, ( configQUEUE_REGISTRY_SIZE * sizeof( QueueRegistryItem_t ) ) );
+}
+
+void tearDown( void )
+{
+    commonTearDown();
+}
+
+void suiteSetUp()
+{
+    commonSuiteSetUp();
+}
+
+int suiteTearDown( int numFailures )
+{
+    return commonSuiteTearDown( numFailures );
+}
+
+/* ==========================  Helper functions =========================== */
+
+static bool helper_find_in_queue_registry( QueueHandle_t xQueue,
+                                           const char * pcQueueName )
+{
+    for( int i = 0; i < configQUEUE_REGISTRY_SIZE; i++ )
+    {
+        if( ( xQueueRegistry[ i ].pcQueueName == pcQueueName ) &&
+            ( xQueueRegistry[ i ].xHandle == xQueue ) )
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static bool helper_find_handle_in_queue_registry( QueueHandle_t xQueue )
+{
+    for( int i = 0; i < configQUEUE_REGISTRY_SIZE; i++ )
+    {
+        if( xQueueRegistry[ i ].xHandle == xQueue )
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/* ==========================  Test Cases =========================== */
+
+/**
+ * @brief Test vQueueAddToRegistry with a NULL QueueHandle_t
+ * @details Verify that a NULL QueueHandle_t results in the string being stored
+ *  to the QueueRegistry and a configASSERT failure.
+ * @coverage vQueueAddToRegistry
+ **/
+void test_vQueueAddToRegistry_null_xQueue( void )
+{
+    const char * pcFakeString = ( char * ) ( BaseType_t ) getNextMonotonicTestValue();
```

Comment:
Maybe rename this as pcFakeStringPtr to be more clear that accessing it is not desired.

Reply:
Done.